queries/nim/folds.scm

```diff
@@ -0,0 +1,44 @@
+[
+  (const_section)
+  (var_section)
+  (let_section)
+  (type_section)
+  (using_section)
+
+  (object_declaration)
+  (tuple_type)
+  (enum_declaration)
+
+  (case)
+  (if)
+  (when)
+  (of_branch)
+  (elif_branch)
+  (else_branch)
+
+  (for)
+  (while)
+  (block)
+  (static_statement)
+
+  (try)
+  (except_branch)
+  (finally_branch)
+
+  (do_block)
+  (call
+    (argument_list
+      (statement_list)))
```

Comment:
In this case, is it better to just capture `argument_list` ? Or is the call supposed to be chained as well

Reply:
That is about a special case in the syntax, where you can pass whole statements (not expressions) as arguments to a function. It is primarily used for macros and templates.

```nim
template_call(arg1:int, arg2:int):
  # here starts the (statement_list) as part of the (argument_list)
  arg3_statement_1 # here would be real statements
  arg3_statement_2
```

So yeah this is specifically about folding the whole call if there is a (statement_list) in the (argument_list)